app/lib/proxy_api/ansible.rb

```diff
@@ -53,5 +53,62 @@ def playbooks(playbooks_names = [])
     rescue *PROXY_ERRORS => e
       raise ProxyException.new(url, e, N_('Unable to get playbooks from Ansible'))
     end
+
+    def repo_information(vcs_url)
+      parse(get("vcs_clone/repo_information?vcs_url=#{vcs_url}"))
+    rescue *PROXY_ERRORS, RestClient::Exception => e
+      raise e unless e.is_a? RestClient::RequestFailed
+      case e.http_code
+      when 400
+        raise Foreman::Exception.new N_('Error requesting repo-info. Check Smartproxy log.')
+      else
+        raise
+      end
+    end
+
+    def list_installed
+      parse(get('vcs_clone/roles'))
+    rescue *PROXY_ERRORS
+      raise Foreman::Exception.new N_('Error requesting installed roles. Check log.')
+    end
+
+    def install_role(repo_info)
+      parse(post(repo_info, 'vcs_clone/roles'))
+    rescue *PROXY_ERRORS, RestClient::Exception => e
+      raise e unless e.is_a? RestClient::RequestFailed
+      case e.http_code
+      when 409
+        raise Foreman::Exception.new N_('A repo with the name %rName already exists.') % repo_info['name']
```

Comment:
Shouldn't it be:
```suggestion
        raise Foreman::Exception.new N_('A repo with the name %rName already exists.') % repo_info['repo_info']['name']
```
And I would suggest adding the `&` operator: `repo_info['repo_info']&.[]('name')`.

Reply:
Indeed. Also, good call on the safe navigation operator. Did not know it could be used that way.